### ISSUE TITLE
Routing System

### ISSUE THREAD

**Issue by tlhunter** [tlhunter/sleekmvc#1]
I've changed my mind... A routing system wouldn't really complicate the framework and prevent it from being beginner friendly.

We should add a routing class, to be defined in the app's bootstrap.php file. This should accept a regular expression for URL matching, and have a system for providing arguments to trigger controllers, actions, and provide arguments.

**Comment by oCuRL** [comment]
In my opinion a routing system would be cool, maybe have another system with the routing system included and another not? Well depends if there is much work todo doing both at the same time i guess.

**Comment by tlhunter** [comment]
I'll add a routing system, and have it so that by default it works with the classical site/controller/action manner.

That way if they don't want to bother with routing they just keep doing the same thing, and if they do, they can add routes and get more flexibility.

**Comment by tlhunter** [comment]
Example code I'm working on. This code goes in index.php before starting Sleek. I'm still debating the syntax though.

```
<?php
// Configure the router class. Add more routes here, above the default one.
// Both :controller and :action are magical, any others are not.
// Paranthesis mean that what is inside is optional.
// Strings starting with : are pattern match groups.

// Here's an example route, where you can browse to /users/1
\Sleek\Request::addRoute('/users/:id', array(
    'controller' => 'user',
    'action' => 'view',

    ':id' => 'numeric',
));

// This is a "catch all" route; in many cases it will be all you need.

// www.example.com/root/
// www.example.com/root/controllerName
// www.example.com/root/controllerName/actionName
// www.example.com/root/controllerName/actionName/something
\Sleek\Request::addRoute('(/:controller(/:action(/:id)))', array(
    'controller' => 'home',
    'action' => 'index',

    ':id' => 'alphanumeric',
));
```

**Comment by tlhunter** [comment]
Fixed in 5c4c7a1d2a7ef154aeae19802f933bfc9853ff11
